src/herder/TransactionQueue.cpp

```diff
@@ -60,8 +60,19 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx)
     // FEE_MULTIPLIER * v2 does not overflow uint128_t because fees are bounded
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
     auto v1 = bigMultiply(newFee, oldNumOps);
-    auto v2 = bigMultiply(oldFee, newNumOps);
-    return v1 >= TransactionQueue::FEE_MULTIPLIER * v2;
+    auto v2 = uint128_t(TransactionQueue::FEE_MULTIPLIER) *
+              bigMultiply(oldFee, newNumOps);
+    std::pair<bool, int64_t> result;
+    result.first = v1 >= v2;
+    if (!result.first)
+    {
+        // Add a fee recommendation for future attempts
```

Comment:
As @MonsieurNicolas pointed out, we shouldn't use the term "fee recommendation" anywhere.